### PR TITLE
add telemetry

### PR DESCRIPTION
**Pull request by SummerSun**

### CODE REVIEW

File: src/Utility.ts [hunk 1]
```diff
@@ -40,6 +41,17 @@ export namespace Utility {
         });
     }
 
+    export function trackTelemetryStep(step: string): void {
+        const session: Session = TelemetryWrapper.currentSession();
+        if (session && session.extraProperties) { session.extraProperties.finishedSteps.push(step); }
```
Comment (andxu) [inline src/Utility.ts]: where is finishedSteps used?

Comment (SummerSun) [reply src/Utility.ts]: TelemetryWrapper track all finished steps in command end event.

File: src/Tomcat/TomcatController.ts [hunk 2]
```diff
@@ -131,6 +147,7 @@ export class TomcatController {
                 vscode.window.showInformationMessage(DialogMessage.serverStopped);
                 return;
             }
+            Utility.trackTelemetryStep(`${restart ? 'restart' : 'stop'}`);
```
Comment (andxu) [inline src/Tomcat/TomcatController.ts]: no need of use ``

File: src/Tomcat/TomcatController.ts [hunk 3]
```diff
@@ -224,8 +251,10 @@ export class TomcatController {
 
         if (pick) {
             if (pick instanceof TomcatServer) {
+                Utility.trackTelemetryStep('select server');
                 return pick;
             } else {
+                Utility.trackTelemetryStep('create server');
                 return await this.createServer();
```
Comment (andxu) [inline src/Tomcat/TomcatController.ts]: create server will break the telemetry sequence.

**Comment by SummerSun** [comment]
#46

**Pull request merged**